sharding/shard.go

```diff
@@ -0,0 +1,208 @@
+package sharding
+
+import (
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/rlp"
+)
+
+// Shard base struct.
+type Shard struct {
+	shardDB *shardKV
+	shardID *big.Int
+}
+
+// MakeShard creates an instance of a Shard struct given a shardID.
+func MakeShard(shardID *big.Int) *Shard {
+	// Swappable - can be makeShardLevelDB, makeShardSparseTrie, etc.
+	shardDB := makeShardKV()
```

Comment:
Shouldn’t this be a function argument? Why does a shard know which database to initialize? It feels a bit out of the responsibilities of a initializing a shard. 

I can imagine that a nontrivial database has many arguments so this method will quickly get out of control.